File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result=[1]*len(nums)
        r=1
        l=1
        n=len(nums)
        for i in range(n):
            result[i]=l
            l=l*nums[i]
        for i in range(n-1,-1,-1):
            result[i]*=r
            r*=nums[i]
        return result


        

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
      
      int l=1,r=1;
      int n=nums.size();
      vector<int> res(n);
      for(int i=0;i<n;i++){
        res[i]=l;
        l=l*nums[i];
      }
      for (int i=n-1;i>=0;i--){
        res[i]*=r;
        r=r*nums[i];
      }
      return res;

    }
};